src/frontends/tensorflow/src/frontend.cpp

```diff
@@ -242,26 +247,30 @@ void FrontEnd::translate_graph(const ov::frontend::InputModel::Ptr& model,
             FRONT_END_GENERAL_CHECK(node_outputs.size() > producer_port_idx,
                                     "Output port with index " + std::to_string(producer_port_idx) + " of " +
                                         producer_name + "node specified as custom output does not exist");
-            results.push_back(std::make_shared<ov::opset8::Result>(node_outputs[producer_port_idx]));
+            auto result_node = std::make_shared<ov::opset8::Result>(node_outputs[producer_port_idx]);
+            set_node_name(model_output_name, result_node);
+            results.push_back(result_node);
         }
     }
     // find all terminal nodes in OV graph to complete list of results
     if (results.empty()) {
         for (const auto& node_output_vector : ng_op_map) {
-            for (const auto& output : node_output_vector.second) {
+            for (size_t output_ind = 0; output_ind < node_output_vector.second.size(); ++output_ind) {
+                auto output = node_output_vector.second[output_ind];
                 if (output.get_target_inputs().empty() &&
                     !std::dynamic_pointer_cast<ov::opset8::Result>(output.get_node_shared_ptr())) {
-                    results.push_back(std::make_shared<ov::opset8::Result>(output));
+                    auto model_output_name =
+                        output.get_node_shared_ptr()->get_name() + ":" + std::to_string(output_ind);
+                    auto result_node = std::make_shared<ov::opset8::Result>(output);
+                    set_node_name(model_output_name, result_node);
+                    results.push_back(result_node);
                 }
             }
         }
     }
 
-    // TODO: reorder results and params according to indices given in RT info (if any)
```

Comment:
is it outdated comment?
why we don't need this now?

Reply:
reverted